ResSimpy/Nexus/DataModels/Network/NexusTargets.py

```diff
@@ -0,0 +1,173 @@
+        Returns:
+            DataFrame: of the properties of the targets through time with each row representing a target.
+        """
+        df_store = obj_to_dataframe(self.__targets)
+        return df_store
+
+    def get_overview(self) -> str:
+        raise NotImplementedError('To be implemented')
+
+    def load(self, surface_file: File, start_date: str, default_units: UnitSystem) -> None:
+        """Calls load target and appends the list of discovered nodes into the NexusTargets object.
+
+        Args:
+            surface_file (File): NexusFile representation of the surface file.
+            start_date (str): Starting date of the run
+            default_units (UnitSystem): Units used in case not specified by surface file.
+
+        Raises:
+            TypeError: if the unit system found in the property check is not a valid enum UnitSystem.
+
+        """
+        new_targets = collect_all_tables_to_objects(surface_file, {'TARGET': NexusTarget},
+                                                    start_date=start_date,
+                                                    default_units=default_units)
+        cons_list = new_targets.get('TARGET')
+        if isinstance(cons_list, dict):
+            raise ValueError('Incompatible data format for additional wells. Expected type "list" instead got "dict"')
+        self._add_to_memory(cons_list)
+
+    def _add_to_memory(self, additional_list: Optional[list[NexusTarget]]) -> None:
+        """Extends the targets object by a list of targets provided to it.
+
+        Args:
+        ----
+            additional_list (Sequence[NexusTarget]): list of nexus targets to add to the targets list.
+
+        Returns:
+        -------
+            None
+        """
+        if additional_list is None:
+            return
+        self.__targets.extend(additional_list)
+
+    def remove(self, target_to_remove: dict[str, None | str | float | int] | UUID) -> None:
+        """Remove a node from the network based on the properties matching a dictionary or id.
+
+        Args:
+            target_to_remove (UUID | dict[str, None | str | float | int]): UUID of the node to remove or a dictionary \
+            with sufficient matching parameters to uniquely identify a node
+
+        """
+        self.__parent_network.get_load_status()
+
+        network_file = self.__parent_network.get_network_file()
+
+        if isinstance(target_to_remove, dict):
+            name = target_to_remove.get('name', None)
+            if name is None:
+                raise ValueError(f'Require target name to remove the target instead got {name=}')
+            name = str(name)
+            target = self.__parent_network.find_network_element_with_dict(name, target_to_remove,
+                                                                          self._network_element_name)
+            target_id = target.id
+        else:
+            target_id = target_to_remove
```

Comment:
this remove function is also slightly simpler now and can be called:

```suggestion
    def remove(self, obj_to_remove: dict[str, None | str | float | int] | UUID) -> None:
        """Remove a wellbore from the network based on the properties matching a dictionary or id.

        Args:
            obj_to_remove (UUID | dict[str, None | str | float | int]): UUID of the wellbore to remove or a dictionary \
            with sufficient matching parameters to uniquely identify a wellbore

        """
        self.__remove_object_operations.remove_object_from_network_main(
            obj_to_remove, self._network_element_name, self.__wellbores)

```

Reply:
implemented as suggested